### DISCUSSION TITLE
tquic_client限制了单连接的最大请求数量

### DISCUSSION THREAD

**Discussion by wdj55556** [Tencent/tquic#240] [General]
我在对tquic_client进行测试时发现，即使设置了max_requests_per_conn这一参数，每个conn发送的请求数量依旧被限制在1000，这个上限是tquic_client设置的吗？还是tquic本身的限制？能否更改呢

**Comment by glime-ai** [comment]
在您提到的情况中，`max_requests_per_conn` 参数应该控制每个连接可以发送的最大请求数。如果您观察到请求数量被限制在1000，这可能是由于TQUIC客户端的默认设置或者是在测试配置中某处设定的限制。

根据提供的文档，没有明确说明TQUIC或其客户端实现中有1000这一硬性限制。因此，这个问题可能源于测试环境的配置或者是TQUIC客户端的某个未文档化的行为。

为了解决这个问题，您可以尝试以下几个步骤：
1. 确认`max_requests_per_conn`参数在命令行中正确设置。
2. 查看是否有其他配置或代码可能影响到连接的请求限制。
3. 如果问题依旧存在，建议在TQUIC的官方支持论坛或通过提交issue的方式询问开发者，看是否TQUIC客户端有未文档化的请求限制。

如果您需要进一步的帮助来调整这个参数或者有关于如何设置的具体问题，可以提供更多的配置详情，以便更准确地诊断问题。

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment]
把使用的具体命令及参数发出来看看

**Reply by wdj55556** [reply]
命令参数如下
`./tquic_client --max-requests-per-thread 0 --log-level off --disable-stateless-reset \
--threads 1 \
--max-concurrent-requests 10 \
--max-requests-per-conn 0 \
https://192.168.50.231:443/index1.html`

**Reply by iyangsj** [reply]
使用的server是 tquic_server还是其它？

**Reply by iyangsj** [reply]
上面的命令会在单线程中创建一个连接，最大并发流10，在该连接上持续发送请求。

例如，以tquic_server 为服务端：
<img width="1721" alt="c" src="https://github.com/Tencent/tquic/assets/780807/f19dfb85-e9ef-49c3-8e31-3c1e14f81b7c">

**Reply by wdj55556** [reply]
我试了一下tquic_server，确实没有这个问题。我对接的是nginx_quic，应该是服务器限制了单连接的最大请求数量。非常感谢您的解答。
